proxy/src/connection.rs

```diff
@@ -165,19 +215,25 @@ impl BoundPort {
                     set_nodelay_or_warn(&socket);
 
                     let tls = match &tls {
-                        Conditional::Some(tls) => match &*tls.config.borrow() {
-                            Conditional::Some(config) =>
-                                Conditional::Some(tls::ConnectionConfig {
-                                    identity: tls.identity.clone(),
-                                    config: config.clone(),
-                                }),
-                            Conditional::None(r) => Conditional::None(*r),
-                        },
+                        Conditional::Some(TlsAccept { sensor, config: conn_config }) =>
+                            match &*conn_config.config.borrow() {
+                                Conditional::Some(config) => {
+                                    let config = tls::ConnectionConfig {
+                                        identity: conn_config.identity.clone(),
+                                        config: config.clone(),
+                                    };
+                                    Conditional::Some((sensor, config))
+                                },
+                                Conditional::None(r) => Conditional::None(*r),
+                            },
                         Conditional::None(r) => Conditional::None(*r),
                     };
                     let conn = match tls {
-                        Conditional::Some(tls) =>
-                            Either::A(ConditionallyUpgradeServerToTls::new(socket, tls)),
+                        Conditional::Some((sensor, config)) => {
+                            let f = ConditionallyUpgradeServerToTls::new(socket, config);
+                            let f = sensor.accept(remote_addr, f);
```

Comment:
I'm skeptical that we need to wrap `f` in this sensor future since the sensor future doesn't seem to add any new *async* logic.